src/api/PluginManager.ts

```diff
@@ -16,18 +16,44 @@ export function isPluginEnabled(plugin: string) {
     return window.Aliucord.settings.get("plugins", {})[plugin] !== false;
 }
 
-export function enablePlugin(plugin: string) {
-    const plugins = window.Aliucord.settings.get("plugins", {});
-    plugins[plugin] = true;
-    window.Aliucord.settings.set("plugins", plugins);
+export async function enablePlugin(plugin: string) {
+    const settingsPlugins = window.Aliucord.settings.get("plugins", {});
+    if (isPluginEnabled(plugin)) throw new Error(`Plugin ${plugin} is already enabled.`);
+
+    settingsPlugins[plugin] = true;
+    window.Aliucord.settings.set("plugins", settingsPlugins);
+
+    const bundleZip = new ZipFile(PLUGINS_DIRECTORY + `${plugin}.zip`, 0, "r");
+    const pluginBuffer = loadPluginBundle(bundleZip);
+
+    bundleZip.close();
+
+    const pluginClass = AliuHermes.run(plugin, pluginBuffer) as typeof Plugin;
+    const enabledPlugin = plugins[plugin] = new pluginClass(plugins[plugin].manifest);
```

Comment:
try/catch around both statements?
not sure how native eval works though

Reply:
Probably a good idea 🚎